doc/storagemond/storagemond-hld.md

```diff
@@ -0,0 +1,377 @@
+
+The goal of the Storage Monitoring Daemon (storagemond) is to provide meaningful metrics for the aforementioned issues and enable streaming telemetry for these attributes so that preventative measures may be triggered in the eventuality of performance degradation.
+
+## 2. Data Collection
+
+We are intrested in the following characteristics that describe various aspects of the disk:
+
+### **2.1 Dynamic Attributes** 
+
+**The following attributes are updated frequently and describe the current state of the disk**
+
+- File System IO Reads
+- File System IO Writes
+- Disk IO Reads
+- Disk IO Writes
+- Reserved Blocks Count
+- Temperature
+- Firmware
+- Health
+
+**Filesystem IO Reads/Writes** - Parsed from the `/proc/diskstats` file, these values correspond to the number of reads and writes successfully carried out in the disk. These values would reset upon reboot.
+
+**Disk IO Reads/Writes** - These fields account for write-amplification and wear-leveling algorithms, and are persistent across reboots and powercycles.
+
+**Reserved Blocks Count** - Reserved blocks are managed by the drive's firmware, and their specific allocation and management may vary between disk manufacturers. The primary purposes of reserved blocks in a disk are:
+
+- **Bad-block replacement:** When the firmware detects a bad block, it can map it to a reserved block and continue using the drive without data loss.
+- **Wear Leveling:** Reserved blocks are used to replace or relocate data from cells that have been heavily used, ensuring that all cells are used evenly. 
+- **Over-Provisioning:** Over-provisioning helps maintain consistent performance and extends the lifespan of the disk by providing additional resources for wear leveling and bad block management.
+- **Garbage collection:** When files are deleted or modified, the old data needs to be erased and marked as available for new data. Reserved blocks can help facilitate this process by providing a temporary location to move valid data from blocks that need to be erased. 
+
+- **Temperature, Firmware, Health** - These fields are self-explanatory
+
+### **2.2 Static Attributes**
+
+**These attributes provide informational context about the Storage disk**
+
+- **Vendor Model**
+- **Serial Number**
+
+These fields are self-explanatory.
+
+
+### **2.3 `storagemond` Daemon Flow**
+
+1. The "storagemond" process will be initiated by the "pmon" Docker container.
+2. Upon initialization, the daemon will gather static information utilizing S.M.A.R.T capabilities through instantiated class objects such as SsdUtil and EmmcUtil. This information will be subsequently updated in the StateDB.
+3. The daemon will parse dynamic attributes also utilizing S.M.A.R.T capabilities via the corresponding class member functions, and update the StateDB on an hourly basis.
```

Comment:
Please make the polling interval configurable through config_db.json

Reply:
Done in latest commit